aptos-move/block-executor/src/executor.rs

```diff
@@ -393,63 +269,68 @@ where
-            // For committed txns with Abort or SkipRest status, early halt BlockSTM.
-            match last_input_output.fee_statement(txn_idx) {
-                Some(fee_statement) => {
-                    accumulated_fee_statement.add_fee_statement(&fee_statement);
-                    txn_fee_statements.push(fee_statement);
-                },
-                None => {
-                    scheduler.halt();
+                counters::update_parallel_txn_gas_counters(txn_fee_statements);
 
-                    Self::update_parallel_block_gas_counters(
-                        accumulated_fee_statement,
-                        (txn_idx + 1) as usize,
-                    );
-                    Self::update_parallel_txn_gas_counters(txn_fee_statements);
-                    info!("[BlockSTM]: Parallel execution early halted due to Abort or SkipRest txn, {} txns committed.", txn_idx + 1);
-                    break;
-                },
-            };
-
-            if let Some(per_block_gas_limit) = maybe_block_gas_limit {
-                // When the accumulated execution and io gas of the committed txns exceeds PER_BLOCK_GAS_LIMIT, early halt BlockSTM.
-                // Storage gas does not count towards the per block gas limit, as we measure execution related cost here.
                 let accumulated_non_storage_gas = accumulated_fee_statement.execution_gas_used()
                     + accumulated_fee_statement.io_gas_used();
-                if accumulated_non_storage_gas >= per_block_gas_limit {
-                    // Set the execution output status to be SkipRest, to skip the rest of the txns.
-                    last_input_output.update_to_skip_rest(txn_idx);
-                    scheduler.halt();
-
-                    Self::update_parallel_block_gas_counters(
-                        accumulated_fee_statement,
-                        (txn_idx + 1) as usize,
-                    );
-                    Self::update_parallel_txn_gas_counters(txn_fee_statements);
-                    counters::EXCEED_PER_BLOCK_GAS_LIMIT_COUNT
-                        .with_label_values(&[counters::Mode::PARALLEL])
-                        .inc();
-                    info!("[BlockSTM]: Parallel execution early halted due to accumulated_non_storage_gas {} >= PER_BLOCK_GAS_LIMIT {}, {} txns committed", accumulated_non_storage_gas, per_block_gas_limit, txn_idx);
-                    break;
-                }
+                info!(
+                    "[BlockSTM]: Parallel execution completed. {} out of {} txns committed. \
+		     accumulated_non_storage_gas = {}, limit = {:?}",
+                    txn_idx + 1,
+                    scheduler.num_txns(),
+                    accumulated_non_storage_gas,
+                    maybe_block_gas_limit,
+                );
+                break;
             }
 
             // Remark: When early halting the BlockSTM, we have to make sure the current / new tasks
```

Comment:
not related but this block of comments seem not in the right place?

Reply:
It is in the right place as a reminder for us :)